sda/context.go

```diff
@@ -57,6 +57,12 @@ func (c *Context) CreateProtocolSDA(name string, t *Tree) (ProtocolInstance, err
 	return pi, err
 }
 
+// ProtocolRegister will sign up a new protocol to this Conode.
+// It returns the ID of the protocol.
+func (c *Context) ProtocolRegister(name string, protocol NewProtocol) (ProtocolID, error) {
```

Comment:
I think a "one sentence" comment on why do we provide this method, since we already have 2 ways in total to register a protocol, would be nice. I think it'll be mostly useful for us, because we forget... :)